plugins/nextcloud-rule-exclusions-before.conf

```diff
@@ -402,6 +402,7 @@ SecRule REQUEST_METHOD "@streq PUT" \
     pass,\
     t:none,\
     nolog,\
+    ctl:ruleRemoveById=920450,\
```

Comment:
This would remove the rule for *all* PUT requests. Try matching against a path, like this rule does with a chain rule. Why do you need to allow restricted headers? Which ones exactly?

Reply:
It was `/accept-charset/` but I did not know how to remove a restricted header from `tx.restricted_headers`.
Unfortunaly I have not observed the FP after the latest update of `OWASP_CRS/4.0.0-rc1`. Nonetheless I would appreciate the oppurtunity if you could tell me how to remove such a header from the a variable. Thanks!

Reply:
See https://github.com/coreruleset/nextcloud-rule-exclusions-plugin/blob/cecb8d59c0335b89aff9d2644a9bbd4603fe898c/plugins/nextcloud-rule-exclusions-before.conf#L52. The rules in this block are essentially copies of the rules in `crs-setup.conf`, with some modification to the default value. The rules need to be copies because of initialization order (i.e., you can't just manipulate the default value here).
If you wanted to allow `Accept-Charset` for *all* requests you would copy that rule to the same block, give it a new ID and remove `accept-charset` from the value. However, I don't think you should do that. Instead, allow `Accept-Charset` in a specific context only, just like you've done. Ideally, you would *only* make the exception for `Accept-Charset` but its not trivial to modify the variable value, so I *think* it's fine in this specific instance to allow all request headers.

Reply:
Okay, i got the FP again. Apparently it only happens every few hours (way less than the sync - and cannot be triggered by sync):

The most elegant way (in my humble opinion) would be to remove the `Accept-Charset` from `tx.restricted_headers`.
If I would set `tx.restricted_headers` to `tx.restricted_headers=/content-encoding/ /proxy/ /lock-token/ /content-range/ /if/ /x-http-method-override/ /x-http-method/ /x-method-override/` I would override any defaults set in `crs-setup.conf`.
So with my knowledge the only way would be to allow every header in changing the rule to this:

    # Allow the data type 'text/calendar'
    SecRule REQUEST_FILENAME "@contains /remote.php/dav/calendars/" \
        "id:9508330,\
        phase:1,\
        pass,\
        t:none,\
        nolog,\
        ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
        setvar:'tx.allowed_request_content_type=%{tx.allowed_request_content_type} |text/calendar| |text/plain|',\
        setvar:'tx.restricted_headers= '"

Would this be the way to go?

Reply:
In general, yes. However, as this happens only in a very specific context I would like to be more restrictive. Do you have the log message for that hit? Then we could analyse for which request exactly the header is required. We could then write a dedicated rule for that which sets the variable to an empty string.

Reply:
I have disabled the rule and hope to trigger that FP again. Ill get back to you.
Thanks again for your time!